nx/source/services/acc.c

```diff
@@ -33,6 +34,113 @@ Service* accountGetService(void) {
     return &g_accSrv;
 }
 
+Result accountGetUserCount(s32* user_count)
+{
+    IpcCommand c;
+    ipcInitialize(&c);
+
+    struct {
+        u64 magic;
+        u64 cmd_id;
+    } *raw;
+
+    raw = ipcPrepareHeader(&c, sizeof(*raw));
+
+    raw->magic = SFCI_MAGIC;
+    raw->cmd_id = 0;
+
+    Result rc = serviceIpcDispatch(&g_accSrv);
+
+    if (R_SUCCEEDED(rc)) {
+        IpcParsedCommand r;
+        ipcParse(&r);
+
+        struct {
+            u64 magic;
+            u64 result;
+            u32 user_count;
+        } *resp = r.Raw;
+
+        rc = resp->result;
+
+        if (R_SUCCEEDED(rc)) {
+            *user_count = resp->user_count;
+        }
+    }
+
+    return rc;
+}
+
+static Result _accountListAllUsers(u128* userIDs)
+{
+    IpcCommand c;
+    ipcInitialize(&c);
+
+    Result rc=0;
+
+    size_t bufsize = ACC_USER_LIST_SIZE * sizeof(u128);
+
+    ipcAddRecvStatic(&c, userIDs, bufsize, 0);
+
+    struct {
+        u64 magic;
+        u64 cmd_id;
+    } *raw;
+
+    raw = ipcPrepareHeader(&c, sizeof(*raw));
+
+    raw->magic = SFCI_MAGIC;
+    raw->cmd_id = 2;
+
+    rc = serviceIpcDispatch(&g_accSrv);
+
+    if (R_SUCCEEDED(rc)) {
+        IpcParsedCommand r;
+        ipcParse(&r);
+
+        struct {
+            u64 magic;
+            u64 result;
+        } *resp = r.Raw;
+
+        rc = resp->result;
+    }
+
+    return rc;
+}
+
+Result accountListAllUsers(u128* userIDs, size_t max_userIDs, size_t *actual_total)
+{
+    Result rc=0;
+
+    u128 *temp_userIDs;
+	temp_userIDs = malloc(sizeof(u128) * ACC_USER_LIST_SIZE);
+
+    rc = _accountListAllUsers(temp_userIDs);
+
+    if (R_SUCCEEDED(rc)) {
+        size_t total_userIDs = 0;
+
+        for (int i = 0; i < ACC_USER_LIST_SIZE; i++) {
+            if (temp_userIDs[i]) {
```

Comment:
Should be the other way around, to match what yls8 requested: `if (!temp_userIDs[i]) break;`

Reply:
That is a lot cleaner, thank you for the suggestion :-)